core/src/components/radio-group/radio-group.tsx

```diff
@@ -51,6 +50,13 @@ export class RadioGroup implements ComponentInterface {
     this.setRadioTabindex(this.value);
   }
 
+  private setRadioChecked = (value: any | undefined) => {
+    const radios = this.getRadios();
+    for (const radio of radios) {
```

Comment:
If we implemented the `ionValueChange` event that amanda suggested in https://github.com/ionic-team/ionic-framework/pull/26223/files#r1014326964, this should't be needed. We could do something like this in `ion-radio` instead: https://github.com/ionic-team/ionic-framework/blob/feature-7.0/core/src/components/accordion/accordion.tsx#L88

Reply:
Yeah I originally spec'd out using `ionValueChange` in the spike, but have been exploring the strategy of adding one-to-many event listeners (`ion-accordion`/`ion-accordion-group`) vs. an API to set the state. 

Do you know of where adding event listeners would perform better vs. a private API method? It seems that adding many listeners (and through our monkey-patched ael for Angular) would be higher memory consumption/bloat.

Reply:
I'm not aware of any benchmarks that we have done in the past regarding this, though my guess would be the performance difference is small. 

Could we keep the architecture consistent for now and then switch in another PR if we find that there's better performance using `setRadioChecked`? That way we can make sure that all components using this architecture get the performance improvements.

It's also worth checking to see if anyone has reported performance issues when checking/unchecking values for components that use the current architecture.

Reply:
I'm good with that. I can create a tech debt ticket to investigate this more. 

Looks like one immediate difference is not using event listeners results in the DOM manipulation in the same frame, where using event listeners introduces a delay. This requires us to use things like `page.waitForChanges()` in our tests around it. Probably not noticeable to a user, but small optimizations we can consider in the future.

Reply:
Ah yeah that's a good point -- the listeners would be async. DX for us would also be a good thing to consider in the tech debt ticket too